tests/ui/structs/default-field-values-support.rs

```diff
@@ -0,0 +1,64 @@
+//@ run-pass
+#![feature(default_field_values)]
+#![allow(unused_variables, dead_code)]
+
+pub struct S;
+
+#[derive(Default)]
+pub struct Foo {
+    pub bar: S = S,
+    pub baz: i32 = 42 + 3,
+}
+
+#[derive(Default)]
+pub enum Bar {
+    #[default]
+    Foo {
+        bar: S = S,
+        baz: i32 = 42 + 3,
+    }
+}
```

Comment:
One sneak feature that is kind of supported now, but we could gate if we want: `#[default] Foo {}` is now accepted, where before it wasn't.